### PR TITLE
Motor size revamp 

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary

SUMMARY: Content, Balance "Add motor_tiny and motor_micro to electric motors. Change weight and volume of electric motors"
#### Purpose of change

Volume and weight of electric motors seemed unrealistic
Usage of motor_small was inconsistent : used both in fan, hair trimmer and electric jackhammer for example.

#### Describe the solution

Changes weight and volume of motors to be closer to some real life equivalent ([forum discussion and references](https://discourse.cataclysmdda.org/t/electric-motor-size-revamp/15932))
Expands the range of motor size by adding two smaller motors : motor_tiny which has the size and power of an electric bike motor and motor_micro which is a very small motor used in recipes only (RC Car, food processor, fan, hair trimmer etc ...)
#### Additional context

[previous PR attempt and discussions](https://github.com/CleverRaven/Cataclysm-DDA/pull/24329)

### CODE REVIEW

File: data/json/items/generic.json [hunk 1]
```diff
@@ -1167,6 +1167,19 @@
...
   },
+  {
+    "type": "GENERIC",
+    "id": "motor_micro",
+    "symbol": ",",
+    "color": "light_gray",
+    "name": "micro motor",
+    "category": "spare_parts",
+    "description": "A very small electric motor like those used in RC cars.  Useful in lots of electronics recipes.",
+    "price": 8000,
```
Comment (Night-Pryanik) [inline data/json/items/generic.json]: Way too much for a RC-car motor.

**Comment by Night-Pryanik** [comment]
For the future: no need to lint the whole json file if this is not a main goal of your PR.

File: data/json/items/generic.json (hunk 1 above)
Comment (Fris0uman) [reply data/json/items/generic.json]: The price is too much ? It's the same as the antenna, what would be a more reasonnable one ?

**Comment by Fris0uman** [comment]
Noted. I just copy-pasted the whole thing in the styling tool just to be sure I wasn't messing anything.

**Comment by Fris0uman** [comment]
So how do I fix what's failing with the travis build ?

**Comment by kevingranade** [comment]
The failing test makes assertions about how far a vehicle can drive on a single charge/refuel, changing weights of motors changes the range (as it should).  We need to check that the changes are reasonable and update the target test values.

**Comment by Fris0uman** [comment]
Here are the references I used, but its just examples of real life equivalent I don't know how representative they are. 

> `enhanced electric motor` is a like a Tesla engine : 780 HP, 175 kg and 250L [Tesla Model S Drive Unit](http://www.evwest.com/catalog/product_info.php?products_id=476)
> 
> `large lectric motor` is like one from a racing car : 420 HP, 90kg and 16L [AM Racing AMR Dual Stack 250-90 AC Motor](http://www.evwest.com/catalog/product_info.php?products_id=300)
> 
> `electric motor`  standard  electric car : 125HP, 68kg and 20L [Curtis 1238e-7621 HPEVS Dual AC-34](http://www.evwest.com/catalog/product_info.php?cPath=8&products_id=170) (I don't know if it's really "standard" or not)
> 
> `small electric motor` motrobike : 15HP , 17kg and 5L [BLDC / PMSM brushless motor HPM-10KW](https://www.miromax.lt/en/m-6/c-39/c-45-brushless_bldc_motor_with_permanent_magnet/product-507-bldc__pmsm_brushless_motor_hpm-10kw_-_nominal_power10kw13kw__134ag174ag___650cm3#to_products)
> 
> `tiny motor` electric bike : 0.7HP , 5.8kg and 0.5L [Front 500w Heavy-Duty Direct-Drive Motor](https://www.ebikekit.com/collections/wheels/products/front-500w-heavy-duty-direct-drive-motor)


Also the problem might come from the fact that I changed the weight but not the `power` because it seemed that the resulting `epower` would be too much for current batteries to handle.

But if the weight counterbalance the increase in `power` we could change `power` without touching `epower` until another PR fixes batteries and `epower` as @mlangsdorf recently [suggested](https://discourse.cataclysmdda.org/t/requesting-feedback-for-a-big-change-in-storage-batteries-and-electric-motors/16370)

**Comment by kevingranade** [comment]
Rebased and merged.

**Pull request closed without merging**